lib/AFS/CellCC/DB.pm

```diff
@@ -66,22 +66,54 @@ _connect($$$$) {
     return $dbh;
 }
 
+# Set timeout to 5 minutes if not specified by the admin
+sub
+_set_db_timeout($) {
+    my ($opts) = @_;
+
+    my $timeout = 300;
+    # In certain situations (e.g. packet loss), requests to the MySQL database
+    # may hang indefinitely. To avoid this undesirable behavior, force the
+    # following types of requests to timeout after a given number of seconds.
+    # If not specified by the admin, set the default timeout to 5 minutes. Note
+    # that mysql-specific directives are ignored by other drivers.
+    if (!exists($opts->{'mysql_connect_timeout'})) {
```

Comment:
to avoid some repeition here, how about:

for my $option ('mysql_connection_timeout', 'mysql_read_timeout', 'mysql_write_timeout') {
    if (!exists($opts->{$option})) {
        $opts->{$option} = $timeout;
    }
}

I believe this could also be shorter as:

$opts->{$option} //= $timeout;

but I think that requires some version of perl that's after our stated requirement of 5.08. and it's less readable; no real reason to do that, I guess.